src/libs/antares/series/series.cpp

```diff
@@ -31,9 +33,31 @@ using namespace Yuni;
 
 namespace Antares::Data
 {
+void TimeSeries::Numbers::registerSeries(const TimeSeries* s)
+{
+    series.push_back(s);
+}
+
+// TODO[FOM] Code duplication
+static bool checkAllElementsIdenticalOrOne(std::vector<uint> w)
+{
+    auto first_one = std::remove(w.begin(), w.end(), 1); // Reject all 1 to the end
+    return std::adjacent_find(w.begin(), first_one, std::not_equal_to<uint>()) == first_one;
+}
+
+bool TimeSeries::Numbers::checkSeriesNumberOfColumnsConsistency() const
+{
+    std::vector<uint> width;
+    for (const auto* s : series)
+        width.push_back(s->numberOfColumns());
 
-TimeSeries::TimeSeries(numbers& tsNumbers) : timeseriesNumbers(tsNumbers)
-{}
+    return checkAllElementsIdenticalOrOne(width);
+}
+
+TimeSeries::TimeSeries(Numbers& tsNumbers) : timeseriesNumbers(tsNumbers)
+{
+    tsNumbers.registerSeries(this);
```

Comment:
We should probably forbid copying of `TimeSeries`, to make sure that this is not called multiple times for the same timeseries.

Reply:
I'd rather prevent copies, but it seems that `AreaScratchpad` must be copyable (used in std::vector), so build fails if I
- Delete copy/move constructor and operator
- Delete copy, keep move constructor and operator.